File: 27-RemoveElement/27-RemoveElement.py
# Last updated: 9/16/2025, 12:53:35 AM
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        n = len(nums)
        p = n-1
        i = n-1
        while(i>=0):
            if(nums[i]==val):
                if i < p:
                    nums[i], nums[p] = nums[p], nums[i]
                p -= 1
            i -= 1
                
        return p+1
        

File: 27-RemoveElement/27-RemoveElement.java
// Last updated: 9/16/2025, 12:53:41 AM
class Solution {
    public int removeElement(int[] nums, int val) {
        int k = 0;
        int p = 0 ;
        int i = nums.length-1;
        while(i>=p)
        {
            if(nums[i] != val)
            {
                k++;
                int temp = nums[i];
                nums[i] = nums[p];
                nums[p] = temp;
                p++;
            }
            else
            i--;
        }

        return k;
    }
}